WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTimezoneDialog.java

```diff
@@ -0,0 +1,338 @@
+
+    public static final String KEY_TIMEZONE = "timezone";
+
+    private class Timezone {
+        private final String label;
+        private final String value;
+        private Timezone(String label, String value) {
+            this.label = label;
+            this.value = value;
+        }
+    }
+
+    private boolean mConfirmed;
+    private String mSelectedTimezone;
+
+    private ListView mListView;
+    private SearchView mSearchView;
+    private View mEmptyView;
+    private View mProgressView;
+
+    private TimezoneAdapter mAdapter;
+    private LayoutInflater mInflater;
+
+    public static SiteSettingsTimezoneDialog newInstance(@NonNull String timezone) {
+        Bundle args = new Bundle();
+        args.putString(KEY_TIMEZONE, timezone);
+
+        SiteSettingsTimezoneDialog dialog = new SiteSettingsTimezoneDialog();
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        mInflater = LayoutInflater.from(getActivity());
+        View view = mInflater.inflate(R.layout.site_settings_timezone_dialog, null);
+
+        mListView = view.findViewById(R.id.list);
+        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Timezone tz = (Timezone) mAdapter.getItem(position);
+                mSelectedTimezone = tz.value;
+                mAdapter.notifyDataSetChanged();
+                hideSearchKeyboard();
+            }
+        });
+
+        mSearchView = view.findViewById(R.id.search_view);
+        mSearchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                if (mAdapter != null) {
+                    mAdapter.getFilter().filter(query);
+                }
+                return true;
+            }
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                if (mAdapter != null) {
+                    mAdapter.getFilter().filter(newText);
+                }
+                return true;
+            }
+        });
+        mSearchView.setEnabled(false);
+        mSearchView.setIconifiedByDefault(false);
+
+        mEmptyView = view.findViewById(R.id.empty_view);
+        mProgressView = view.findViewById(R.id.progress_view);
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.Calypso_AlertDialog);
+        builder.setPositiveButton(android.R.string.ok, this);
+        builder.setNegativeButton(R.string.cancel, this);
+        builder.setView(view);
+
+        mSelectedTimezone = getArguments().getString(KEY_TIMEZONE);
+        requestTimezones();
+
+        return builder.create();
+    }
+
+    private void requestTimezones() {
```

Comment:
Having a dialog do a request & handle response feels very weird to me. I really think Dialogs should be dumb UI components. I'd personally prefer to save the response json somewhere as static resource rather than have network code here, but it's up to you.